packages/lesswrong/server/editor/make_editable_callbacks.ts

```diff
@@ -133,9 +140,32 @@ function addEditableCallbacks<T extends DbObject>({collection, options = {}}: {
       const userId = currentUser._id
       const editedAt = new Date()
       const changeMetrics = htmlToChangeMetrics("", html);
+      const isFirstDebatePostComment = 'debate' in doc
+        ? (!!doc.debate && fieldName === 'contents')
+        : false;
+      // Debate posts don't actually have any post contents, since the contents are managed as comments
+      const originalContents: DbRevision["originalContents"] = doc[fieldName].originalContents
+
+      // We need to validate that we'll be able to successfully create the comment in the updateFirstDebateCommentPostId callback
+      // If we can't, we'll be stuck with a malformed debate post with no comments
+      if (isFirstDebatePostComment) {
+        const createFirstCommentParams: CreateMutatorParams<DbComment> = {
+          collection: Comments,
+          document: {
+            userId,
+            contents: doc[fieldName],
+            debateComment: true,
+          },
+          context,
+          currentUser,
+        };
+
+        await validateCreateMutation(createFirstCommentParams);
```

Comment:
Just to check, this throws if it's not a valid comment?

Reply:
Yep